test/reports/test-report-manager.js

```diff
@@ -0,0 +1,290 @@
+const mockgoose = require('mockgoose');
+const mongoose = require('mongoose');
+const assert = require('assert');
+mockgoose(mongoose);
+
+const reportManager = require('../../src/reports/report-manager');
+const reportModels = require('../../src/reports/report-models');
+const Report = reportModels.Report;
+const EventReport = reportModels.EventReport;
+const OtherReport = reportModels.OtherReport;
+
+describe('reportManager', () => {
+  // Mockgoose doesn't start the mock until after 'mongoose.connect()' is
+  // called.
+  // TODO(jmtaber129): Find a version of mockgoose that works and doesn't
+  // print to console each time 'mongoose.connect()' is called.
+  before((done) => { mongoose.connect('', done); });
+
+  // Clear out the mocked database before each test case.
+  beforeEach((done) => {
+    mockgoose.reset();
+    done();
+  });
+
+  after((done) => { mongoose.unmock(done); });
+
+  describe('#createReport', () => {
+    it('should pass a created event report to the callback', (done) => {
+      const eventReport = {
+        type: 'event',
+        approvalStatus: true,
+        student: 'John Doe',
+        event: 'someEventUid',
+      };
+      
+      reportManager.createReport(eventReport, {}, (err, createdReport) => {
+        assert.equal(err, null);
+        assert.equal(createdReport.type, 'EventReport');
+        assert.equal(createdReport.approvalStatus, false);
+        assert.equal(createdReport.student, eventReport.student);
+        assert.equal(createdReport.event, eventReport.event);
+
+        Report.findById(createdReport.id, (err, foundReport) => {
+          assert.equal(err, null);
+          assert.equal(foundReport.type, 'EventReport');
+          assert.equal(foundReport.approvalStatus, false);
+          assert.equal(foundReport.student, eventReport.student);
+          assert.equal(foundReport.event, eventReport.event);
+          done();
+        });
+      });
+    });
+
+    it('should pass a created other report to the callback', (done) => {
+      const otherReport = {
+        type: 'other',
+        approvalStatus: true,
+        student: 'John Doe',
+        category: 'Some category',
+        datetime: 'Apr 04 2017',
+        location: 'EC203',
+        description: 'Some report description',
+      };
+      
+      reportManager.createReport(otherReport, {}, (err, createdReport) => {
+        assert.equal(err, null);
+        assert.equal(createdReport.type, 'OtherReport');
+        assert.equal(createdReport.approvalStatus, false);
+        assert.equal(createdReport.student, otherReport.student);
+        assert.equal(createdReport.category, otherReport.category);
+        assert.equal(
+            new Date(createdReport.datetime).getTime(),
```

Comment:
I found that I can use ```chai-moment``` (https://www.npmjs.com/package/chai-moment) to verify that the ```createdReport.datetime``` object is the same as the ```otherReport.datetime``` string without creating new Date objects.